test/weak_ptr_test.cpp

```diff
@@ -801,7 +801,7 @@ void test()
     }
 
     {
-        boost::shared_ptr<X> sp(static_cast<X*>(0));
+        boost::shared_ptr<X> sp(static_cast<X*>(BOOST_NULLPTR));
```

Comment:
static_cast<T*>(0) would never trigger the warning, even with -Wzero-as-null-pointer-constant